examples/thoughts/ext/Tag.js

```diff
@@ -0,0 +1,30 @@
+import React from 'react'
+import ColorHash from 'color-hash'
+import { Link } from 'react-router-dom'
+
+import './Tag.css'
+
+const getClasses = (href, isActive, isDeletable) => `${href ? 'link' : ''} ${isActive ? 'active' : ''} ${isDeletable ? 'deletable' : ''}`
+const getTagColor = (name) => new ColorHash().hex(name)
+
+const Tag = ({
+  name,
+  isActive,
+  onClick,
+  href
+}) => {
+  const isDeletable = !!onClick
```

Comment:
Hmmm I get why is this so, but it sounds somewhat stretched. What I mean is that assuming that if tag is not clickable it has to be deletable sounds like a too strong assumption. I would suggest we instead make isDeletable an explicit prop. And we can throw runtime error if both onClick and isDeletable are provided.

Reply:
I think you might have misinterpreted what this is doing. The assumption here is that if `onClick` is set, the tag should have deletable styles. The double `!!` indicates a double negative, converting `onClick` from a function to a boolean in this statement.

That being said. This is easily misinterpreted and could do with something a bit more explicit. I'll define an `isDeletable` prop specifically for styling. We may want a tag that we can click later but that we don't want deletable styles on.

Reply:
Ah yes I missed that there are two `!`, but the idea remains the same -> that maybe we should not tie deletability to clickiness. But we agree on that so great!